pkg/vault/test/foundry/BufferVaultPrimitive.t.sol

```diff
@@ -449,143 +449,169 @@ contract BufferVaultPrimitiveTest is BaseVaultTest {
 
         BufferAndLPBalances memory afterBalances = _measureBuffer();
 
         assertEq(
             afterBalances.buffer.dai,
-            beforeBalances.buffer.dai + underlyingAmountIn,
+            beforeBalances.buffer.dai + expectedUnderlyingAmountIn,
             "Buffer DAI balance is wrong"
         );
         assertEq(
             afterBalances.buffer.waDai,
-            beforeBalances.buffer.waDai + wrappedAmountIn,
+            beforeBalances.buffer.waDai + expectedWrappedAmountIn,
             "Buffer waDAI balance is wrong"
         );
 
-        assertEq(afterBalances.vault.dai, beforeBalances.vault.dai + underlyingAmountIn, "Vault DAI balance is wrong");
+        assertEq(
+            afterBalances.vault.dai,
+            beforeBalances.vault.dai + expectedUnderlyingAmountIn,
+            "Vault DAI balance is wrong"
+        );
         assertEq(
             afterBalances.vault.waDai,
-            beforeBalances.vault.waDai + wrappedAmountIn,
+            beforeBalances.vault.waDai + expectedWrappedAmountIn,
             "Vault waDAI balance is wrong"
         );
 
         assertEq(
             afterBalances.vaultReserves.dai,
-            beforeBalances.vaultReserves.dai + underlyingAmountIn,
+            beforeBalances.vaultReserves.dai + expectedUnderlyingAmountIn,
             "Vault Reserve DAI balance is wrong"
         );
         assertEq(
             afterBalances.vaultReserves.waDai,
-            beforeBalances.vaultReserves.waDai + wrappedAmountIn,
+            beforeBalances.vaultReserves.waDai + expectedWrappedAmountIn,
             "Vault Reserve waDAI balance is wrong"
         );
 
-        assertEq(afterBalances.lp.dai, beforeBalances.lp.dai - underlyingAmountIn, "LP DAI balance is wrong");
-        assertEq(afterBalances.lp.waDai, beforeBalances.lp.waDai - wrappedAmountIn, "LP waDAI balance is wrong");
+        assertEq(afterBalances.lp.dai, beforeBalances.lp.dai - expectedUnderlyingAmountIn, "LP DAI balance is wrong");
+        assertEq(
+            afterBalances.lp.waDai,
+            beforeBalances.lp.waDai - expectedWrappedAmountIn,
+            "LP waDAI balance is wrong"
+        );
 
         assertEq(
-            lpSharesBeforeAdd + lpSharesAdded,
+            lpSharesBeforeAdd + lpSharesToAdd,
             vault.getBufferOwnerShares(IERC4626(address(waDAI)), lp),
             "LP Buffer shares is wrong"
         );
-        assertEq(lpSharesAdded, underlyingAmountIn + waDAI.previewRedeem(wrappedAmountIn), "Issued shares is wrong");
+
+        // Since underlying and wrapped amount in are rounded up, it can introduced rounding issues in favor of the
+        // vault (`bufferInvariantDelta - 2e6 < sharesToAdd < bufferInvariantDelta`).
+        uint256 bufferInvariantDelta = expectedUnderlyingAmountIn + waDAI.previewRedeem(expectedWrappedAmountIn);
+        assertApproxEqAbs(lpSharesToAdd, bufferInvariantDelta - 1e6, 1e6, "Issued shares is wrong");
```

Comment:
I'd consider using `Ge` / `Le` to show that the error only goes one way.

Reply:
I usually used approxEq and Ge or Le, but I found out that this test covers both cases. Is it a bad pattern?